internal/resources/metal/virtual_circuit/resource.go

```diff
@@ -37,6 +37,12 @@ func Resource() *schema.Resource {
 				Description: "UUID of Connection where the VC is scoped to",
 				ForceNew:    true,
 			},
+			"virtual_circuit_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "UUID of an existing VC to configure. Used in the case of shared interconnections where the VC has already been created.",
```

Comment:
Is `virtual_circuit_id` somewhat redundant, able to be inferred, from the required `connection_id`?

`virtual_circuit_id` can be confusing compared to the resource's `id`. I think users will naturally expect `virtual_circuit_id` to be set when a new VC is created and be confused when the value is empty.

Some alternative ways to approach/name this: (just thinking out loud)
* `behaviors { connection_prepared = true }`
* `use_shared_connection_virtual_circuit = true`

If the upstream extended the service to offer HA pairs in shared virtual circuits (similar to what Network Edge has introduced in the service over time), the `virtual_circuit_id` approach would be preferable. If we can't think of an alternative name, perhaps the description could add something like: "`virtual_circuit_id` is an optional parameter, not to be confused with the computed `id` attribute.".

Reply:
I believe there can be multiple virtual circuits associated with the shared connection, so I don't think we can infer which one to use. 

What about a name like `shared_connection_virtual_circuit_id` in addition to some details in the description help to help make it more clear?

Reply:
There's precedent for the `virtual_circuit_id` approach in the `metal_port` resource, which accepts a `port_id` attribute.  That usage is a little different because a port can't be created directly, but I'm not sure we need much more than a longer description that more explicitly calls out that `virtual_circuit_id` is not the `id` attribute.

If we start getting into behavior flags and long attribute names to differentiate this, that brings me back to either having separate `metal_virtual_circuit` and `metal_shared_virtual_circuit` resources or moving the BGP settings to a `metal_virtual_circuit_bgp_config` resource.

Reply:
Looking at this again, IMO the current description of the `virtual_circuit_id` attribute is sufficient for now, but the requirement to provide a `connection_id` in all cases could be potentially confusing.  I think we only need `connection_id` when creating dedicated connection VCs, in which case we should update the `connection_id` attribute definition as follows:

```
"connection_id": {
				Type:        schema.TypeString,
				Optional:    true,
                                 Computed: true,
                                 ExactlyOneOf: []string{"connection_id", "virtual_circuit_id"},
				Description: "UUID of Connection where the VC is scoped to.  Only used for dedicated connections",
				ForceNew:    true,
			}
```